jdaviz/configs/default/plugins/model_fitting/fitting_backend.py

```diff
@@ -100,7 +100,10 @@ def _fit_1D(initial_model, spectrum, run_fitter, window=None):
 
     """
     if run_fitter:
-        weights = 'unc' if spectrum.uncertainty else None
+        if spectrum.uncertainty and np.all(spectrum.uncertainty.array != 0):
```

Comment:
Do we want a Snackbar warning that uncertainty is ignored or users won't care?

Reply:
I think as long as the logic is to only ignore ALL zeros, that the user won't care.  If we change this to `np.any(spectrum.uncertainty.array <= 0)` or similar, then I think we'd definitely need some sort of warning?

Reply:
If I read this correctly, *any* zero in the uncertainty would cause it to be ignored.

Reply:
Or maybe you want `not np.allclose(spectrum.uncertainty.array, 0)`?

Reply:
ah you're right, I misread which was the `if` and which was the `else`.

What about:
```
if spectrum.uncertainty:
    if np.all(spectrum.uncertainty.array == 0):
        weights = None
    else:
        weights = 'unc' # will likely still raise astropy error if any 0s, nans, infs... maybe eventually we want a warning for that case?
else:
    weights = None
```

Reply:
We have a ticket to allow the user to toggle whether or not they want to use the uncertainties, maybe we can add a warning as part of that work if the user explicitly says to use the uncertainties.

Reply:
Whoops, I only saw the first comment when I replied. I think checking all rather than any zeros is probably fine for now.